tools/src/h5diff/ph5diff_main.c

```diff
@@ -279,11 +279,12 @@ h5diff_exit(int status)
             phdiff_dismiss_workers();
             MPI_Barrier(MPI_COMM_WORLD);
         }
-        MPI_Finalize();
-        status = EXIT_SUCCESS; /* Reset exit status, since some mpiexec commands generate output on failure
-                                  status */
     }
 
+    MPI_Finalize();
+
+    status = EXIT_SUCCESS; /* Reset exit status, since some mpiexec commands generate output on failure status */
```

Comment:
Since the CMake testing can filter out stderr now, we likely should change this back to having ph5diff return the correct values instead of always returning 0, but it's simplest to leave this as is for now.